sdk/dapp-sdk/src/ui.ts

```diff
@@ -19,45 +26,32 @@ class GatewayUi {
         switch (walletType) {
             case 'remote':
                 // Focus the existing popup if it's already open
-                if (this.popup === undefined || this.popup.closed) {
-                    popupHref(new URL(userUrl), {
-                        target: this.identifier,
-                    }).then((window) => {
+                if (
+                    this.popup === undefined ||
+                    this.popup.closed ||
+                    this.lastUrl !== userUrl
+                ) {
+                    popupHref(new URL(userUrl)).then((window) => {
                         this.popup = window
+                        this.lastUrl = userUrl
                     })
                 } else {
                     this.popup.focus()
                 }
+
                 return this.popup
-                break
             case 'extension': {
                 const msg: SpliceMessage = {
                     type: WalletEvent.SPLICE_WALLET_EXT_OPEN,
                     url: userUrl,
                 }
                 window.postMessage(msg, '*')
-                break
             }
         }
     }
 
     close() {
-        if (this.popup) {
-            this.popup.close()
-        } else {
-            // If the page was refreshed, the popup may still be open, but the WindowProxy
-            // reference is lost. Attempt to close it by grabbing a new handle and closing immediately.
-            const existing = window.open(
-                '',
-                this.identifier,
-                'popup width=100,height=100'
-            )
-
-            if (existing && !existing.closed) {
-                existing.close()
-            }
-        }
-
```

Comment:
this approach would sometimes lead to a flash of an empty popup, which I thought looked bad. 

Instead, I decided to automatically close the popup if the dApp page is refreshed / closed via the `beforeunload` event